core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/ai-content-prompt.component.ts

```diff
@@ -14,42 +12,31 @@ interface AIContentForm {
 @Component({
     selector: 'dot-ai-content-prompt',
     templateUrl: './ai-content-prompt.component.html',
-    styleUrls: ['./ai-content-prompt.component.scss']
+    styleUrls: ['./ai-content-prompt.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class AIContentPromptComponent {
-    isFormSubmitting = false;
+export class AIContentPromptComponent implements OnInit {
+    vm$: Observable<AiContentPromptState> = this.aiContentPromptStore.vm$;
 
     @ViewChild('input') private input: ElementRef;
-    @Output() formSubmission = new EventEmitter<boolean>();
-    @Output() aiResponse = new EventEmitter<string>();
 
     form: FormGroup<AIContentForm> = new FormGroup<AIContentForm>({
         textPrompt: new FormControl('', Validators.required)
     });
 
-    constructor(private aiContentService: AiContentService) {}
+    constructor(private readonly aiContentPromptStore: AiContentPromptStore) {}
+
+    ngOnInit() {
+        this.aiContentPromptStore.open$.subscribe((open) => {
```

Comment:
Do we need to kill this subscribe on destroy?

Reply:
it looks that is not needed, but added to prevent leaks.